src/Controls/src/Core/HandlerImpl/RoundRectangle.Impl.cs

```diff
@@ -36,5 +37,39 @@ public override PathF GetPath()
 
 			return path;
 		}
+
+		PathF IRoundRectangle.InnerPathForBounds(Rect viewBounds, float strokeWidth)
+		{
+			if (HeightRequest < 0 && WidthRequest < 0)
+			{
+				Frame = viewBounds;
```

Comment:
The Frame should not be set here; this will cause re-layouts. And GetInnerPath should not use the Width and Height properties, because those will not be set before the first layout pass has completed. See #12326 and #12356 for more info.